### PR TITLE
ISPN-9013 Partition Handling enabled toggle should set merge-policy=NONE

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-9013

Added deprecation javadoc and fix log message.

### CODE REVIEW

File: core/src/main/java/org/infinispan/util/logging/Log.java [hunk 1]
```diff
@@ -1591,7 +1591,7 @@ TimeoutException coordinatorTimeoutWaitingForView(int expectedViewId, int curren
    IllegalStateException getConflictsStateTransferInProgress(String cacheName);
 
    @LogMessage(level = WARN)
-   @Message(value = "The partition handling 'enable' attribute has been deprecated. Please update your configuration file to use the 'type' attribute instead", id = 464)
+   @Message(value = "The partition handling 'enable' attribute has been deprecated. Please update your configuration to use 'when-split' instead", id = 464)
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/util/logging/Log.java]: s/enable/enabled/

**Comment by danberindei** [comment]
Fixed @tristantarrant

**Comment by wburns** [comment]
I am fine with the changes, but thinking about this more. If we are doing this for backwards compatibility shouldn't the default be `NONE` no matter what? @tristantarrant what do you think?

**Comment by tristantarrant** [comment]
I think we should have better defaults win over blind backwards compatibility. As long as a user can configure the old behaviour, I'm fine with it.

**Pull request merged**